src/topology/separation.lean

```diff
@@ -16,6 +16,58 @@ variables {α : Type u} {β : Type v} [topological_space α]
 
 section separation
 
+/--
+`separate` is a predicate on pairs of `finset`s of a topological space.  It holds if the two
+`finset`s are contained in disjoint open sets.
+-/
+def separate : finset α → finset α → Prop :=
+  λ (s t : finset α), ∃ U V : (set α), (is_open U) ∧ is_open V ∧
+  (∀ a : α, a ∈ s → a ∈ U) ∧ (∀ a : α, a ∈ t → a ∈ V) ∧ disjoint U V
+
+@[symm] lemma separate.symm {s t : finset α} : separate s t → separate t s :=
+begin
+  rintros ⟨U, V, oU, oV, aU, bV, UV⟩,
+  exact ⟨V, U, oV, oU, bV, aU, disjoint.symm UV⟩
+end
+
+lemma separate.comm (s t : finset α) : separate s t ↔ separate t s :=
+⟨separate.symm, separate.symm⟩
+
+lemma separate.empty_right : ∀ {a : finset α}, separate a ∅ :=
+λ a, ⟨_, _, is_open_univ, is_open_empty, λ a h, mem_univ a, λ a h, by cases h, disjoint_empty _⟩
```

Comment:
you can still get rid of this lambda, by moving the argument left of the colon. Also, I would make the finset argument explicit, so that it can easily be provided if it can't be infered from the context.